consensus/obcpbft/broadcast.go

```diff
@@ -67,35 +73,71 @@ func (b *broadcaster) Wait() {
 	b.closed.Wait()
 }
 
-func (b *broadcaster) unicastOne(msg *pb.Message, dest uint64, wait chan bool) {
+func (b *broadcaster) drainerSend(dest uint64, send *sendRequest, printedValidatorNotFound bool) bool {
 	defer func() {
 		b.closed.Done()
-		wait <- true
 	}()
+	h, err := getValidatorHandle(dest)
+	if err != nil {
+		if !printedValidatorNotFound {
+			logger.Warningf("could not get handle for replica %d", dest)
+		}
+		time.Sleep(time.Second)
+		send.done <- false
+		return true
+	}
 
-	select {
-	case <-b.closedCh:
-		return
-	default:
+	if printedValidatorNotFound {
+		logger.Infof("Found handle for replica %d", dest)
+		printedValidatorNotFound = false
 	}
 
-	h, err := getValidatorHandle(dest)
+	err = b.comm.Unicast(send.msg, h)
 	if err != nil {
-		logger.Warningf("could not get handle for replica %d", dest)
+		logger.Warningf("could not send to replica %d: %v", dest, err)
+		send.done <- false
 	} else {
-		err = b.comm.Unicast(msg, h)
+		send.done <- true
 	}
-	if err != nil {
-		logger.Debugf("could not send to replica %d: %v", dest, err)
+
+	return false
+}
+
+func (b *broadcaster) drainer(dest uint64) {
+	printedValidatorNotFound := false
+
+	for {
 		select {
 		case <-b.closedCh:
```

Comment:
the order of the case statements is a bit confusing.  I'd put the main work case on top.